crates/forge/src/main.rs

```diff
@@ -55,6 +55,27 @@ struct Args {
     /// Clean forge cache directory
     #[arg(short, long)]
     clean_cache: bool,
+
+    /// Cores used for parallel tests execution
+    #[arg(long, value_parser = validate_cores_number)]
+    cores: Option<usize>,
+}
+
+fn validate_cores_number(val: &str) -> Result<usize> {
+    let parsed_val: usize = val
+        .parse()
+        .map_err(|_| anyhow::anyhow!("Failed to parse '{}' as usize", val))?;
+    if parsed_val == 0 {
+        bail!("Number of cores must be greater than 0");
+    }
+    let cores_approx = available_parallelism()?.get();
+    if parsed_val > cores_approx {
+        bail!(
+            "Number of cores must be less than or equal to the number of cores available on the machine ({})",
```

Comment:
```suggestion
            "Number of cores must be less than or equal to the number of cores available on the machine = {}",
```
We usually format these error like this